### PR TITLE
Small edits to the build your own workshop dropdown

### PR DESCRIPTION
**Pull request by hannahbergam**
These are the two edits:

1. Filter course offerings further so levelbuilder permissions don't show work-in-progress (read: sandbox) courses
2. Edit the scroll functionality of the dropdown so the max height is 300px

Screenshot 2024-07-24 at 4 37 54 PM

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-2110

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx [hunk 1]
```diff
@@ -1212,16 +1219,13 @@ export class WorkshopForm extends React.Component {
...
                       <div
-                        className={classnames('dropdown-menu')}
+                        className={'dropdown-menu'}
+                        style={styles.dropdown}
                         aria-labelledby="dropdownMenuButton"
                       >
                         <ul style={styles.listItems}>
                           {Object.values(ALL_PL_TOPICS).map(topic => (
-                            <li
-                              className="dropdown-item"
```
Comment (hannahbergam) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:1221]: this className wasn't doing anything, so this is cleanup

File: dashboard/app/controllers/course_offerings_controller.rb [hunk 2]
```diff
@@ -37,7 +37,9 @@ def quick_assign_course_offerings
 
   def self_paced_pl_course_offerings
     return head :bad_request unless current_user
-    offerings = CourseOffering.assignable_course_offerings(current_user).select {|co| co.get_participant_audience == 'teacher' && co.instruction_type == 'self_paced'}
+    offerings = CourseOffering.assignable_course_offerings(current_user).filter do |co|
+      co.get_participant_audience == 'teacher' && co.any_version_is_in_published_state? && co.instruction_type == 'self_paced' && co.header.present?
```
Comment (TurnerRiley) [inline dashboard/app/controllers/course_offerings_controller.rb:41]: What is `co.header.present?` checking?

**Review by TurnerRiley — APPROVED** [review]
Looks great! Just had a question for my own understanding

File: dashboard/app/controllers/course_offerings_controller.rb (hunk 2 above)
Comment (hannahbergam) [reply dashboard/app/controllers/course_offerings_controller.rb:41]: It's the way we filter out course offerings that were created in LB but never meant to appear on curriculum quick assign (it represents a label 'self-paced' that these all have). For example, there are a couple sandbox course offerings and such that don't have headers. I believe the published state will also filter these out, but I wanted to be consistent with this marketing field from LB to match what we have in quick assign!

Comment (TurnerRiley) [reply dashboard/app/controllers/course_offerings_controller.rb:41]: Thank you gotcha!

**Pull request merged**